ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -922,18 +921,18 @@ IndexValue markAsPermanent(StoreKey id, FileSpan fileSpan, long operationTimeMs)
    * @return the {@link IndexValue} of the ttl update record
    * @throws StoreException if there is any problem writing the index record
    */
-  private IndexValue markAsPermanent(StoreKey id, FileSpan fileSpan, MessageInfo info, long operationTimeMs,
-      short lifeVersion) throws StoreException {
+  IndexValue markAsPermanent(StoreKey id, FileSpan fileSpan, MessageInfo info, long operationTimeMs, short lifeVersion)
+      throws StoreException {
     validateFileSpan(fileSpan, true);
     boolean hasLifeVersion = IndexValue.hasLifeVersion(lifeVersion);
     IndexValue value = findKey(id);
     if (value == null && info == null) {
       throw new StoreException("Id " + id + " not present in index " + dataDir, StoreErrorCodes.ID_Not_Found);
     }
     short retrievedLifeVersion = value == null ? info.getLifeVersion() : value.getLifeVersion();
```

Comment:
In which case the `value` could be null ?

Reply:
Honestly i don't know it was here for a while then.